Userland/Services/WebContent/InputSource.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include "AK/Debug.h"
+#include <AK/HashTable.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/String.h>
+
+namespace Web::WebDriver {
+
+// InputState has lists of InputSources, but the pointer input source constructor
+// depends on InputState, so the definitions are circular
+// forward declare InputState here
+class InputState;
+
+// https : // w3c.github.io/webdriver/#input-sources
+enum class InputSourceType { none,
+    key,
+    pointer,
+    wheel };
+// this enum is for the types of devices, which make up the subtype field of the PointerInputSource class
+enum class PointingDeviceSubtype { mouse,
+    pen,
+    touch };
```

Comment:
As far as I can tell from the spec, these are both string inputs which are used once to decide which substep to run. So, making them enums might not be worth it.

As for naming, we name enum members like types, so it'd be:
```c++
enum class PointingDeviceSubtype {
    Mouse,
    Pen,
    Touch
};
````